File: csharp/Ice/multicast/DiscoverI.cs
// Copyright (c) ZeroC, Inc.

using Demo;

internal class DiscoverI(Ice.ObjectPrx obj) : DiscoverDisp_
{
    public override void
    lookup(DiscoverReplyPrx reply, Ice.Current current)
    {
        try
        {
            reply.reply(_obj);
        }
        catch (Ice.LocalException)
        {
            // Ignore
        }
    }

    private Ice.ObjectPrx _obj = obj;
}

File: java/Ice/multicast/DiscoverI.java
//
// Copyright (c) ZeroC, Inc. All rights reserved.
//

import com.zeroc.demos.Ice.multicast.Demo.*;

public class DiscoverI implements Discover
{
    DiscoverI(com.zeroc.Ice.ObjectPrx obj)
    {
        _obj = obj;
    }

    @Override
    public void lookup(DiscoverReplyPrx reply, com.zeroc.Ice.Current current)
    {
        try
        {
            reply.reply(_obj);
        }
        catch(com.zeroc.Ice.LocalException ex)
        {
            // Ignore
        }
    }

    private com.zeroc.Ice.ObjectPrx _obj;
}
